lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -337,6 +350,33 @@ static Class getClassProp(Schema schema, String prop) {
     ARRAY_CLASSES.put(boolean.class, boolean[].class);
   }
 
+  /**
+   * It returns false for non-string-maps because Avro writes out such maps
+   * as an array of records. Even their JSON representation is an array.
+   */
+  protected boolean isMap(Object datum) {
```

Comment:
NP. I have changed the name back. I think we probably do not need isStringMap being called from isMap in ReflectData if that is what isMap solely does.
The current implementation of isMap kind of has that already built in it.
No?
*return (datum instanceof Map) && !isNonStringMap(datum);*